modules/core/src/main/java/org/apache/ignite/internal/manager/IgniteComponent.java

```diff
@@ -17,15 +17,19 @@
 
 package org.apache.ignite.internal.manager;
 
+import java.util.concurrent.CompletableFuture;
+
 /**
  * Common interface for ignite components that provides entry points for component lifecycle flow.
  */
 public interface IgniteComponent {
     /**
      * Starts the component. Depending on component flow both configuration properties listeners, meta storage watch registration, starting
      * thread pools and threads goes here.
+     *
+     * @return Future that will be completed when the asynchronous part of the start is processed.
      */
-    void start();
+    CompletableFuture<Void> start();
```

Comment:
Do we really need this method right now? I mean, it is quiet easy to add it to `IgniteComponents` as `deafult` method, but in this PR we do not override this method in any class, so it will be just unused method. I would add it when somebody really needs this method in they logic.

Reply:
I believe that we need it. Because without it, general async components infrastructure isn't complete. Basically that means that it's not possible to run safe async code just because it's not possible to verify whether dependent component is ready.

Reply:
ok, I'll add this default method to `IgniteComponents`

Reply:
Just in case, simple default method addition won't be enough of course.

Reply:
Could you please explain then, what do you expect? Implementation of this method for all components?